service/topology_coordinator.cc

```diff
@@ -752,6 +753,87 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         case global_topology_request::cleanup:
             co_await start_cleanup_on_dirty_nodes(std::move(guard), true);
             break;
+        case global_topology_request::keyspace_rf_change: {
+            while (true) {
+                auto tmptr = get_token_metadata_ptr();
+                sstring ks_name = *_topo_sm._topology.new_keyspace_rf_change_ks_name;
+                std::unordered_map<sstring, sstring> new_rf_per_dc = *_topo_sm._topology.new_keyspace_rf_change_rf_per_dc;
+                std::map<sstring, sstring> new_rf_per_dc_map{new_rf_per_dc.begin(), new_rf_per_dc.end()};
+                sstring req_uuid = *_topo_sm._topology.global_request_id;
+                std::vector<canonical_mutation> updates;
+                sstring error;
+                auto& ks = _db.find_keyspace(ks_name);
+                tablet_replica_calculation_result new_data{locator::tablet_map{8}, tablet_replica_calculation_status::success};
+
+                for (const auto& table : ks.metadata()->tables()) {
+                    auto repl_strategy = ks.get_replication_strategy_ptr()->maybe_as_tablet_aware();
+                    try {
+                        new_data = co_await service::calculate_tablet_replicas_for_new_rf(
+                                repl_strategy, table, tmptr, new_rf_per_dc_map);
+                    } catch(const std::exception& e) {
+                        new_data.status = tablet_replica_calculation_status::other;
+                        error = e.what();
+                        rtlogger.error("{}", error);
+                    }
+
+                    if (new_data.status != tablet_replica_calculation_status::success) {
+                        rtlogger.error("Couldn't process global_topology_request::keyspace_rf_change, status: {}, rf per dc map: {}",
```

Comment:
why would `new_tablet_map` be valid (or interesting) on error?
if at all, the `new_rf_per_dc` would be more interesting to print.

Reply:
Also, `error` might be unset on this path, if new_data failed with no exception.
Is this intended?

Reply:
ping

Reply:
Changed to logging new_rf_per_dc.
2 paths of errors handling have been merged, because status codes have been removed, so now `error` is only set when an actual error happens.